src/http/errors.ts

```diff
@@ -96,6 +117,25 @@ function isPlainObject(obj: Any): obj is Record<string, unknown> {
   return typeof obj === 'object' && obj !== null && !Array.isArray(obj)
 }
 
+export function groqParseMessage(res: QueryParseError, tag?: string | null) {
+  const {query, start, end, description} = res.error
+  const lineStart = query.slice(0, start).lastIndexOf('\n') + 1
+  const lineNumber = (query.slice(0, lineStart).match(/\n/g) || []).length + 1
+  const line = query.slice(lineStart, query.indexOf('\n', lineStart))
```

Comment:
These calculations may be slightly off if there is no `\n` in the string, since `lastIndexOf()` would return -1.

Note that `start` and `end` might point to a broader region across multiple lines.